src/fabric/src/fabric2_db.erl

```diff
@@ -808,12 +808,19 @@ validate_docid(<<"_design/">>) ->
 validate_docid(<<"_local/">>) ->
     throw({illegal_docid, <<"Illegal document id `_local/`">>});
 validate_docid(Id) when is_binary(Id) ->
-    MaxLen =
-        case config:get("couchdb", "max_document_id_length", "infinity") of
-            "infinity" -> infinity;
-            IntegerVal -> list_to_integer(IntegerVal)
+    ConfigDocIdLength = config:get_integer("replicator", "max_document_id_length", ?DOC_ID_LIMIT_BYTES),
```

Comment:
This should be "couchdb" I believe

Reply:
There's the typo! Thank you. See https://github.com/apache/couchdb/pull/3632/commits/468123da973d22ff9d4e91477b1f48568cde80b9.